pkg/controller/manager/manager_controller.go

```diff
@@ -240,20 +240,24 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
-	// Create the cert if doesn't exist. If the cert exists, check that the cert
-	// has the expected DNS names. If the cert doesn't and the cert is managed by the
-	// operator, the cert is recreated and returned. If the invalid cert is supplied by
-	// the user, set the component degraded.
-	svcDNSNames := dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain)
-	svcDNSNames = append(svcDNSNames, "localhost")
-	certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
-	tlsSecret, err = utils.EnsureCertificateSecret(
-		render.ManagerTLSSecretName, tlsSecret, render.ManagerSecretKeyName, render.ManagerSecretCertName, certDur, instance.GetUID(), svcDNSNames...,
-	)
+	// If the manager certpair secret exists but is not managed by the operator,
+	// then skip checking its DNS names.
+	if tlsSecret != nil && utils.IsOwnedByUID(tlsSecret, instance.GetUID()) {
+		// Create the cert if doesn't exist. If the cert exists, check that the cert
+		// has the expected DNS names. If the cert doesn't and the cert is managed by the
+		// operator, the cert is recreated and returned. If the invalid cert is supplied by
+		// the user, set the component degraded.
```

Comment:
We won't check a cert created by the user inside this block because of the IsOwnedByUID above.